### ISSUE TITLE
Unable to use

### ISSUE THREAD

**Issue by oneplusreddy** [hlf20010508/telegram-onedrive#28]
Hello Dev,  I did everything which mentioned in guide, but my process stuck with this message in telegram "Code received, logining..." after sending telegram code. Please let me know where I am doing wrong. Let me know if you need more details. Please see logs.

telegram-onedrive  |  INFO received an update referencing an unknown peer, treating as gap
telegram-onedrive  | 2024-12-28 05:30:12.697  grammers_session::message_box /usr/local/cargo/git/checkouts/grammers-689e30b82f69dcd5/b733d70/lib/grammers-session/src/message_box/mod.rs:395
telegram-onedrive  |  INFO received an update referencing an unknown peer, but cannot find out who
telegram-onedrive  | 2024-12-28 05:30:12.697  grammers_session::message_box /usr/local/cargo/git/checkouts/grammers-689e30b82f69dcd5/b733d70/lib/grammers-session/src/message_box/mod.rs:391
telegram-onedrive  |  INFO received an update referencing an unknown peer, treating as gap

**Comment by oneplusreddy** [comment]
services:
  telegram-onedrive:
    container_name: telegram-onedrive
    image: hlf01/telegram-onedrive
    restart: always
    volumes:
      - telegram-onedrive-session:/session
      - telegram-onedrive-logs:/logs
      # - /path/to/*.crt:/ssl/server.crt
      # - /path/to/*.key:/ssl/server.key
    ports:
      - 8883:8080
    environment:
      - port=8883
      # - trace_level=info
      # - worker_num=5
      - server_uri=https://127.0.0.1
      - reverse_proxy=true
      - tg_bot_token=74900X72X9:AAGXBSqXAUeyA6nexdzxY5ORv3s4kc4wNS4
      - tg_api_id=130XXXXX
      - tg_api_hash=83XX62dc0X1feXeXd2da363d87c91c53
      - tg_user_phone=+9198864XXXXX
      # - tg_user_password=xxxxxxxx
      # - tg_user_name=xxxxxxxx
      - od_client_id=X4aXa4cX-1126-4Xd9-a6e9-542bc7Xf7706
      - od_client_secret=XXXXX~Y1RFg4NXXXXX~U_XF~XXXXX_7UiUJ4XXXX
      - od_root_path=/Torbox
      # - auto_delete=true

volumes:
  telegram-onedrive-session:
  telegram-onedrive-logs:

**Comment by hlf20010508** [comment]
You don't need to specify port in the environment.

Delete `- port=8883`.

**Comment by oneplusreddy** [comment]
Since I am trying from long time, currently I am blocked. Will try this after my block period over.

- authorization error: request error: rpc error 420: FLOOD_WAIT caused by auth.sendCode (value: 27772)

**Comment by hlf20010508** [comment]
You mean you requested tg code too much?

**Comment by hlf20010508** [comment]
And your `server_uri` should be `https://127.0.0.1:8883`.

Redirect uri on azure portal should be `https://127.0.0.1:8883/auth`.

If still not working, don't try too many times, report here.

**Comment by oneplusreddy** [comment]
> You mean you requested tg code too much?

yes

**Comment by oneplusreddy** [comment]
> And your `server_uri` should be `https://127.0.0.1:8883`.
> 
> Redirect uri on azure portal should be `https://127.0.0.1:8883/auth`.
> 
> And `reverse_proxy` should be false, since you're running on local machine, I don't think you're using nginx or others.
> 
> If still not working, don't try too many times, report here.

No I am not using localhost, I am using server with easypanel.

**Comment by hlf20010508** [comment]
Then the uri should not be `127.0.0.1`. It should be the public address of your server.

**Comment by oneplusreddy** [comment]
> Then the uri should not be `127.0.0.1`. It should be the public address of your server.

Sir, I tried even with my server name but still same issue, so tried with that. Is it possible to Install through docker, what env do I need to add ? below mentioned is enough ? 

port=8883
server_uri=https://apps-onedrive.major.easypanel.host
reverse_proxy=true
tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
tg_api_id=1308XXXX
tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
tg_user_phone=+9198864XXXXX
od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
od_root_path=/Torbox

**Comment by hlf20010508** [comment]
Don't worry, the problem is just the uri.

Let's deploy locally first, to refresh your tg code limitation.

```yml
services:
  telegram-onedrive:
    container_name: telegram-onedrive
    image: hlf01/telegram-onedrive
    restart: always
    volumes:
      - telegram-onedrive-session:/session
      - telegram-onedrive-logs:/logs
    ports:
      - 8883:8080
    environment:
      - server_uri=https://127.0.0.1:8883
      - tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
      - tg_api_id=1308XXXX
      - tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
      - tg_user_phone=+9198864XXXXX
      - od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
      - od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
      - od_root_path=/Torbox

volumes:
  telegram-onedrive-session:
  telegram-onedrive-logs:
```

And set redirect uri on azure portal to `https://127.0.0.1:8883/auth`.

This should at least allow you to run locally.

**Comment by oneplusreddy** [comment]
> Don't worry, the problem is just the uri.
> 
> Let's deploy locally first, to refresh your tg code limitation.
> 
> ```yaml
> services:
>   telegram-onedrive:
>     container_name: telegram-onedrive
>     image: hlf01/telegram-onedrive
>     restart: always
>     volumes:
>       - telegram-onedrive-session:/session
>       - telegram-onedrive-logs:/logs
>     ports:
>       - 8883:8080
>     environment:
>       - server_uri=https://127.0.0.1:8883
>       - tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
>       - tg_api_id=1308XXXX
>       - tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
>       - tg_user_phone=+9198864XXXXX
>       - od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
>       - od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
>       - od_root_path=/Torbox
> 
> volumes:
>   telegram-onedrive-session:
>   telegram-onedrive-logs:
> ```
> 
> And set redirect uri on azure portal to `https://127.0.0.1:8883/auth`.
> 
> This should at least allow you to run locally.

Will try and let you know later today. Thanks for your quick response, you are a great DEV. :)

**Comment by oneplusreddy** [comment]
> > Don't worry, the problem is just the uri.
> > Let's deploy locally first, to refresh your tg code limitation.
> > ```yaml
> > services:
> >   telegram-onedrive:
> >     container_name: telegram-onedrive
> >     image: hlf01/telegram-onedrive
> >     restart: always
> >     volumes:
> >       - telegram-onedrive-session:/session
> >       - telegram-onedrive-logs:/logs
> >     ports:
> >       - 8883:8080
> >     environment:
> >       - server_uri=https://127.0.0.1:8883
> >       - tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
> >       - tg_api_id=1308XXXX
> >       - tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
> >       - tg_user_phone=+9198864XXXXX
> >       - od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
> >       - od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
> >       - od_root_path=/Torbox
> > 
> > volumes:
> >   telegram-onedrive-session:
> >   telegram-onedrive-logs:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And set redirect uri on azure portal to `https://127.0.0.1:8883/auth`.
> > This should at least allow you to run locally.
> 
> Will try and let you know later today. Thanks for your quick response, you are a great DEV. :)

I have a low config laptop, it is not supporting docker to install.
Could you please help me to install this in my [easypanel ](https://easypanel.io/) server. How many ways I try I am getting error "received an update referencing an unknown peer, but cannot find out who"

**Comment by hlf20010508** [comment]
> received an update referencing an unknown peer, but cannot find out who

You don't need to pay attention to this log, it has nothing to do with this project.

I never used easypanel before, but I'll try registering and deploying.

**Comment by hlf20010508** [comment]
> I never used easypanel before, but I'll try registering and deploying.

I can't find a way to register and login. Does this need to self-host?

**Comment by oneplusreddy** [comment]
> > I never used easypanel before, but I'll try registering and deploying.
> 
> I can't find a way to register and login. Does this need to self-host?

Easypanel is like cpanel but through Easypanel you can easily deploy dockers. Shall I provide my server details to check why its not working please ?

**Comment by hlf20010508** [comment]
Yeah you can show more details if you can.

Try this

```yml
services:
  telegram-onedrive:
    container_name: telegram-onedrive
    image: hlf01/telegram-onedrive
    restart: always
    volumes:
      - telegram-onedrive-session:/session
      - telegram-onedrive-logs:/logs
    ports:
      - 8883:8080
    environment:
      - server_uri=https://apps-onedrive.major.easypanel.host
      - reverse_proxy=true
      - tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
      - tg_api_id=1308XXXX
      - tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
      - tg_user_phone=+9198864XXXXX
      - od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
      - od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
      - od_root_path=/Torbox

volumes:
  telegram-onedrive-session:
  telegram-onedrive-logs:
```

And set redirect uri on azure portal to https://apps-onedrive.major.easypanel.host/auth

**Comment by oneplusreddy** [comment]
> Yeah you can show more details if you can.
> 
> Try this
> 
> ```yaml
> services:
>   telegram-onedrive:
>     container_name: telegram-onedrive
>     image: hlf01/telegram-onedrive
>     restart: always
>     volumes:
>       - telegram-onedrive-session:/session
>       - telegram-onedrive-logs:/logs
>     ports:
>       - 8883:8080
>     environment:
>       - server_uri=https://apps-onedrive.major.easypanel.host
>       - reverse_proxy=true
>       - tg_bot_token=74900XXXXX:AAG7BSqUAUeyA6nexdzxY5ORv3s4kc4wNS4
>       - tg_api_id=1308XXXX
>       - tg_api_hash=XXXXXXX31fe4e1d2da363d87c91c53
>       - tg_user_phone=+9198864XXXXX
>       - od_client_id=a4a2a4c8-XXXX-42d9-XXXX-542bc75f7706
>       - od_client_secret=XXXXX~XXXX4NFubsa~U_XF~Q5KKwO_7UiUXXXX
>       - od_root_path=/Torbox
> 
> volumes:
>   telegram-onedrive-session:
>   telegram-onedrive-logs:
> ```
> 
> And set redirect uri on azure portal to https://apps-onedrive.major.easypanel.host/auth

Yes I did this but no luck. Could you please share your email, so that I can send my Easypanel server details?

**Comment by hlf20010508** [comment]
Add my tg [@hlf01](https://t.me/hlf01)

**Comment by hlf20010508** [comment]
Fixed in 4.0.2
